### DISCUSSION TITLE
fluid-synth formula problems to keep up with upstream

### DISCUSSION THREAD

**Discussion by pedrolcl** [Homebrew/discussions#1577] [Tap maintenance and Homebrew development]
fluid-synth formula is currently stuck in v2.1.8 because the latest intents to bump failed to build some dependents (see Homebrew/homebrew-core/pull/74428 and Homebrew/homebrew-core/pull/77948).

Currently, the following formulae depend on fluid-synth:

https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/carla.rb
https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/dosbox-x.rb
https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/scummvm.rb
https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/dosbox-staging.rb
https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/audacious.rb
https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/mpd.rb
https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/csound.rb

Two casks (VMPK and dmidiplayer) also depend on it.

The upstream release v2.2.0 states in its [release notes](https://github.com/FluidSynth/fluidsynth/releases/tag/v2.2.0) that

> This release breaks ABI compatibility! Refer to the [API docs](https://www.fluidsynth.org/api/RecentChanges.html#NewIn2_2_0) for details.

It is expected that some projects adopt the new major version sooner than others. My suggestion is to create a new versioned formula, fluid-synth@2.2 to handle this case.

**Comment by pedrolcl** [comment]
@carlocab @cho-m

Upstream fluidsynth releases belong to 3 branches:

1.x deprecated, no formulae on Homebrew depend on it anymore. Latest release is [v1.1.11](https://github.com/FluidSynth/fluidsynth/releases/tag/v1.1.11)
2.0 (including 2.1) 7 formulae on Homebrew depend on it. Latest release is [v2.1.9](https://github.com/FluidSynth/fluidsynth/releases/tag/v2.1.9) 
2.2.x the latest [v2.2.1](https://github.com/FluidSynth/fluidsynth/releases/tag/v2.2.1) The next VMPK and dmidiplayer casks will use it if available in Homebrew

**Comment by carlocab** [comment]
Which ones of the dependent formulae need `fluid-synth` 2.1? It seems to me that only `scummvm` needs it.

**Comment by carlocab** [comment]
`fluid-synth` was updated in https://github.com/Homebrew/homebrew-core/pull/81146.
